lib/rspec/matchers/built_in/compound.rb

```diff
@@ -162,8 +162,13 @@ def initialize(actual, matcher_1, matcher_2)
 
             inner, outer = order_block_matchers
 
+            returned = nil
             @match_results[outer] = outer.matches?(Proc.new do |*args|
-              @match_results[inner] = inner.matches?(inner_matcher_block(args))
+              p = Proc.new { |*args2|
+                returned = inner_matcher_block(args).call(*args2)
+              }
```

Comment:
This needs to take care of ruby2 keywords (e.g. marked as such, not having keywords itself) so that they get passed through, I'm also a bit nervous about `returned` why isn't it just implict from the assignment?